src/test/java/org/opentripplanner/routing/fares/OrcaFareServiceTest.java

```diff
@@ -91,6 +91,21 @@ public void calculateFareWithNoFreeTransfer() {
         calculateFare(rides, Fare.FareType.electronicYouth, 0f + DEFAULT_RIDE_PRICE_IN_CENTS + 175f);
     }
 
+    /**
+     * Check to make sure the fare by leg is calculated properly for a trip with two rides.
+     */
+    @Test
+    public void calculateFareByLeg() {
+        List<Ride> rides = Arrays.asList(
+            getRide(KITSAP_TRANSIT_AGENCY_ID, 0),
+            getRide(COMM_TRANS_AGENCY_ID, 2)
+        );
+        Fare fare = new Fare();
+        orcaFareService.populateFare(fare, null, Fare.FareType.electronicRegular, rides, null);
+        Assertions.assertEquals(349, rides.get(0).fareComponents.get(Fare.FareType.electronicRegular).price.getCents());
```

Comment:
Can you also add an assertion that `isTransfer` is false for the first transit leg, and true for the second leg, please?

Reply:
Good idea.